datapoints.go

```diff
@@ -134,3 +130,22 @@ func (dps *DataPoints) Len() int {
 
 	return len(dps.datapoints)
 }
+
+// filter returns a new DataPoints structure consisting of only those
+// points where filterFunc is true
+func (dps *DataPoints) filter(filterFunc func(*DataPoint) bool) *DataPoints {
```

Comment:
Only real reason is my instinctual conservatism. I think that the filter idiom can be pretty powerful for 'listy' containers. I can export it if you'd like; NBD.